bundles/mapping/layerswipe/handler/SwipeToolHandler.js

```diff
@@ -0,0 +1,65 @@
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+
+class UIHandler extends StateHandler {
+    constructor (tool) {
+        super();
+        this.tool = tool;
+        this.sandbox = tool.getSandbox();
+        this.setState({
+            autoStart: false,
+            noUI: false
+        });
+    };
+
+    init (pluginConfig) {
+        this.updateState({
+            ...pluginConfig.conf,
+            autoStart: pluginConfig?.state?.active,
+            noUI: pluginConfig?.conf?.noUI
+        });
+    }
+
+    syncState () {
+        // required to sync state to plugin on startup and restore the state of the embedded map
+        const { autoStart, noUI } = this.getState();
+        this.setAutoStart(autoStart);
+        this.setHideUI(noUI);
+    }
+
+    clearState () {
+        // plugin is created again on startup, so it's state doesn't need to be cleared
+        this.setState({
+            autoStart: false,
+            noUI: false
+        });
+    }
+    /*
+    onLayersChanged () {
+        // we could do setEnabled(false) here if Oskari.getSandbox().getMap().getLayers() is < 2
```

Comment:
We should add a ticket to implement this: if user removes layers while in publisher -> disable this if there aren't enough layers to compare? And maybe check that this can't be enabled with just one layer?